extras/linked-token/test/IpcTokenReplica.t.sol

```diff
@@ -54,9 +59,37 @@ contract IpcTokenReplicaTest is Test, IntegrationTestBase {
         vm.prank(address(rootNativeSubnetActor));
         registerSubnetGW(DEFAULT_COLLATERAL_AMOUNT, address(rootNativeSubnetActor), rootGateway);
 
-        controller = new LinkedTokenController(gateway, controllerSubnetUSDC, replicaSubnetName);
-        replica.initialize(address(controller));
-        controller.initialize(address(replica));
+        //set up controller with proxy
+        LinkedTokenController initialControllerImplementation = new LinkedTokenController();
+        TransparentUpgradeableProxy transparentProxyController = new TransparentUpgradeableProxy(
+            address(initialControllerImplementation),
+            address(this),
+            ""
+        );
+        controller = LinkedTokenController(address(transparentProxyController));
+
+        //set up replica with proxy
+        LinkedTokenReplica initialReplicaImplementation = new LinkedTokenReplica();
+        TransparentUpgradeableProxy transparentProxyReplica = new TransparentUpgradeableProxy(
+            address(initialReplicaImplementation),
+            address(this),
+            ""
+        );
+        replica = LinkedTokenReplica(address(transparentProxyReplica));
+
+        // initialize controller & replica
+
+        controller.initialize(gateway, controllerSubnetUSDC, replicaSubnetName, address(replica));
+
+        replica.initialize(
+            gateway,
+            controllerSubnetUSDC,
+            controllerSubnet,
+            address(controller),
+            REPLICA_TOKEN_NAME,
+            REPLICA_TOKEN_SYMBOL,
+            REPLICA_TOKEN_DECIMALS
+        );
     }
```

Comment:
We don't need to call `setLinkedContract()` here?

Reply:
I polished the test here e3bb723b

Also 1974d9a3 I renamed the tests to LinkedToken* instead of IpcToken*. These two files have tests for just the replica and controller. And didn't need the two way connection between the contracts so the setLinkedContract wasn't. The MultiSubnet test file has the linked contract tests.